### PR TITLE
Fix issues related to RER series creation

### PR DESCRIPTION
**Pull request by lucatume**
Tickets: https://central.tri.be/issues/79347 and https://central.tri.be/issues/79353

This PR contains fixes needed by PRO RER code and fixes (and tests) to the common code.

### CODE REVIEW

File: src/Tribe/Date_Utils.php [hunk 1]
```diff
@@ -1134,6 +1134,38 @@ public static function unescape_date_format( $date_format ) {
...
+				'sixth'   => 6,
+				'seventh' => 7,
+				'eighth'  => 8,
+				'ninth'   => 9,
+				'tenth'   => 10,
+			);
+
+			$key = strtolower( $ordinal );
+
+			$number =  isset($map[$key]) ? $map[$key] : false;
```
Comment (borkweb) [inline src/Tribe/Date_Utils.php]: Spacing: `$map[ $key ]` vs `$map[$key]`

**Comment by borkweb** [comment]
After the spacing changes, this is good for a merge! :+1:

File: src/Tribe/Date_Utils.php (hunk 1 above)
Comment (bordoni) [inline src/Tribe/Date_Utils.php]: Weird spacing and missing spaces. Should we use `Tribe__Utils__Array`?

Comment (lucatume) [inline src/Tribe/Date_Utils.php]: I'd say no: no need to pull in a dependency when we are not gaining much in term of readability.

**Pull request merged**